### PR TITLE
implement expression context in Wasm

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>
Implement CEL context in Wasm context.
Harden the getters to handle empty stream info.
Drop request_protocol since it's in upstream now.
Fixes getProperty returning Ok for missing properties.

/assign @jplevyak 

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/common/wasm/context.h [hunk 1]
```diff
@@ -290,6 +293,15 @@ class Context : public Logger::Loggable<Logger::Id::wasm>,
   virtual WasmResult grpcSend(uint32_t token, absl::string_view message,
                               bool end_stream); // stream only
 
+  // CEL evaluation
+  virtual std::vector<const google::api::expr::runtime::CelFunction*>
+  FindFunctionOverloads(absl::string_view) const override {
```
Comment (jplevyak) [inline source/extensions/common/wasm/context.h]: Functions start with a lower case letter.

Comment (kyessenov) [reply source/extensions/common/wasm/context.h]: This is an override for a google-style declaration.

**Comment by kyessenov** [comment]
@bianpengyuan Can you try to update to this and let me know if this fixes https://github.com/istio/proxy/pull/2548 ASAN failure? Note there might be a gap with request.protocol since it may not have been picked up from upstream yet.

**Comment by bianpengyuan** [comment]
Sure, will post back.

**Comment by kyessenov** [comment]
Peter's problem seems to be caused by a different issue (uninitialized struct).
Can we merge this? I don't want bad SHAs before this one and after refactor to make it to istio/proxy. It's painful to debug.

**Comment by kyessenov** [comment]
@PiotrSikora merge this one, too?

**Pull request merged**